### DISCUSSION TITLE
Arch

### DISCUSSION THREAD

**Discussion by metalgearsloth** [space-wizards/space-station-14#21850] [Code announcements]
Arch was tentatively shipped for 181.0.0  
arch tl;dr
- completely changes how components are stored from dictionaries to arrays
- makes multi-component changes O(1) instead of O(n) when the engine API is updated
- should be faster once we take advantage of a decent ECS framework
- minimal content changes, mainly around nullability of EntityManager methods.

For those of you who weren't around when we removed IEntity expect bugs and patience.

**Comment by Brandon-Huu** [comment]
How will this affect the economy

**Reply by deltanedas** [reply]
since this is shitpost tree, btw i use

**Reply by deathride58** [reply]
STONKS UP
